apps/web/src/views/Lottery/components/Slider.tsx

```diff
@@ -91,6 +143,87 @@ const SlideContent = ({ title, date }) => {
 const Slider = () => {
   const [activeItemIndex, setActiveItemIndex] = useState(0)
   const { isDark } = useTheme()
+  const {
+    t,
+    currentLanguage: { locale },
+  } = useTranslation()
+  const dispatch = useAppDispatch()
+  const {
+    currentLotteryId,
+    lotteriesData,
+    currentRound: { status },
+  } = useLottery()
+  const [latestRoundId, setLatestRoundId] = useState(null)
+  const [selectedRoundId, setSelectedRoundId] = useState('')
+  const [selectedLotteryNodeData, setSelectedLotteryNodeData] = useState(null)
+  const timer = useRef(null)
+
+  const numRoundsFetched = lotteriesData?.length
+
+  const items = useMemo(() => {
+    return lotteriesData
+      ?.filter(lottery => lottery.status.toLowerCase() !== 'open')
+      .map(lottery => ({
+        title: `Round ${lottery.id}`,
+        date: getDrawnDate(locale, lottery.endTime),
+        lotteryData: { ...lottery },
+      }))
+  }, [lotteriesData, locale])
+
+  useEffect(() => {
+    if (currentLotteryId) {
+      const currentLotteryIdAsInt = currentLotteryId ? parseInt(currentLotteryId) : null
+      const mostRecentFinishedRoundId =
+        status === LotteryStatus.CLAIMABLE ? currentLotteryIdAsInt : currentLotteryIdAsInt - 1
+      setLatestRoundId(mostRecentFinishedRoundId)
+      setSelectedRoundId(mostRecentFinishedRoundId.toString())
+    }
+  }, [currentLotteryId, status])
+
+  useEffect(() => {
+    setSelectedLotteryNodeData(null)
+
+    const fetchLotteryData = async () => {
+      const lotteryData = await fetchLottery(selectedRoundId)
+      const processedLotteryData = processLotteryResponse(lotteryData)
+      setSelectedLotteryNodeData(processedLotteryData)
+    }
+
+    timer.current = setInterval(() => {
+      if (selectedRoundId) {
+        fetchLotteryData()
+      }
+      clearInterval(timer.current)
+    }, 1000)
+
+    return () => clearInterval(timer.current)
+  }, [selectedRoundId, currentLotteryId, numRoundsFetched, dispatch])
+
+  const handleInputChange = event => {
+    const {
+      target: { value },
+    } = event
+    if (value) {
+      setSelectedRoundId(value)
```

Comment:
Assign a variable and then call `setSelectedRoundId`, don't call `setSelectedRoundId` multiple times

Reply:
removed this parent function since it was unused